### PR TITLE
Update development environment

### PR DESCRIPTION
**Pull request by pellared**
## Why

It was not updated for a while. I also hope that GitHub Codespaces can make contributions easier.

## What

- Update the existing development environment docs
- Add support for GitHub Codespaces for ease of use and contributions
- Increase the `sleep` in examples as SQL Server is often not ready - I plan to refine the example in a separate PR

## Tests

- In GitHub Codespaces
- MacBook

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -4,41 +4,45 @@
...
+### Linux
 
-When using [C# for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp),
-follow these steps to set `omnisharp.useGlobalMono` to `never`:
+- cmake, make, gcc, clang, clang++
 
-1. Go to `File` -> `Preferences` -> `Settings` -> `Extensions` -> `C# Configuration`.
-2. Change `Omnisharp: Use Global Mono` to `never`.
-3. Restart OmniSharp by pressing `F1` and selecting `OmniSharp: Restart OmniSharp`.
+On macOS it should be enough to
```
Comment (pjanotti) [inline docs/developing.md:37]: Leftover

File: dev/codespaces-init.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -ex
+
+# install Nuke globally
+dotnet tool install Nuke.GlobalTool --global
```
Comment (pjanotti) [inline dev/codespaces-init.sh:5]: Can we use `dotnet tool restore` instead?

File: docs/developing.md [hunk 3]
```diff
@@ -4,41 +4,45 @@
...
 with Windows and Visual Studio preinstalled.
 
-### Linux and macOS
+### macOS
 
-- cmake, make, gcc, clang, and clang++
-- [Docker](https://docs.docker.com/engine/install/)
-- [Docker Compose](https://docs.docker.com/compose/install/)
+- Run: `xcode-select --install`
+- [Visual Studio 2022 for Mac (17.4)](https://visualstudio.microsoft.com/vs/mac/)
```
Comment (pjanotti) [inline docs/developing.md:30]: Is this really required? Or is it here because brings dependencies in one go?

File: dev/codespaces-init.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+# install .NET 6.0 SDK
```
Comment (Kielek) [inline dev/codespaces-init.sh:7]: I think that we need 7.0, at least for tests.

File: examples/Makefile [hunk 5]
```diff
@@ -4,7 +4,7 @@ run:
 	chmod 777 log
 	docker-compose build
 	docker-compose up -d service
-	sleep 10 # wait for bootstrap
+	sleep 30 # wait for bootstrap
```
Comment (pjanotti) [inline examples/Makefile:7]: Almost saying that it should ping or health check the service per earlier conversation on other PR.

File: docs/developing.md [hunk 6]
```diff
@@ -4,41 +4,45 @@
 
 On all platforms, the minimum requirements are:
 
-- Git client and command line tools
+- [Docker](https://docs.docker.com/engine/install/)
+- [Docker Compose](https://docs.docker.com/compose/install/)
 - [.NET 6.0 SDK](https://dotnet.microsoft.com/download/dotnet/6.0)
```
Comment (Kielek) [inline docs/developing.md:9]: Is it really needed?

File: dev/codespaces-init.sh (hunk 2 above)
Comment (pellared) [reply dev/codespaces-init.sh:5]: OK

File: docs/developing.md (hunk 3 above)
Comment (pellared) [reply docs/developing.md:30]: > Or is it here because brings dependencies in one go?

I will remove it 😉

File: examples/Makefile (hunk 5 above)
Comment (pellared) [reply examples/Makefile:7]: I want to address it in another PR 😉

File: docs/developing.md (hunk 6 above)
Comment (pellared) [reply docs/developing.md:9]: yes. I was getting some errors. I can check later if we can get rid of them somehow.

Comment (pellared) [reply docs/developing.md:9]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1723

File: dev/codespaces-init.sh (hunk 2 above)
Comment (pellared) [reply dev/codespaces-init.sh]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1720/commits/74d0e370c8d4c97978e2ae98b32422586756bd2c

File: docs/developing.md (hunk 1 above)
Comment (pellared) [reply docs/developing.md]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1720/commits/ea749237f709cda9ca90155fb4e6563ab371c308

File: docs/developing.md (hunk 3 above)
Comment (pellared) [reply docs/developing.md]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1720/commits/ea749237f709cda9ca90155fb4e6563ab371c308

**Review by Kielek — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by RassK — APPROVED** [review]

**Pull request merged**